ocaml/xapi/repository_helpers.ml

```diff
@@ -1235,7 +1208,90 @@ let get_updates_from_list_updates repositories =
   in
   updates
 
-let get_update_in_json ~update_ids ~installed_pkgs (new_pkg, repo) =
+let validate_latest_updates ~latest_updates ~accumulative_updates =
+  List.map
+    (fun (pkg, repo) ->
+      match List.assoc_opt pkg accumulative_updates with
+      | Some uid ->
+          (pkg, Some uid, repo)
+      | None ->
+          warn "Not found update ID for update %s" (Pkg.to_fullname pkg) ;
+          (pkg, None, repo)
+      )
+    latest_updates
+
+let prune_by_latest_updates latest_updates pkg uid =
+  let open Pkg in
+  let is_same_name_arch pkg1 pkg2 =
+    pkg1.name = pkg2.name && pkg1.arch = pkg2.arch
+  in
+  match
+    List.find_opt (fun (pkg', _) -> is_same_name_arch pkg pkg') latest_updates
+  with
+  | Some (pkg', repo) ->
+      if
+        Applicability.gt pkg.epoch pkg.version pkg.release pkg'.epoch
+          pkg'.version pkg'.release
```

Comment:
Maybe `repository_helpers.ml` should have a function for comparing packages

Reply:
Indeed. I can move the comparison from Applicability to Pkg.